### PR TITLE
Support if/else route definitions in `no-shadow-route-definition` rule

### PR DESCRIPTION
**Pull request by raido**
This PR is an attempt support if else statements for route definitions.

We are building prefix key from block statements ranges that are within given if statement block and use it for routeMap lookup.

Closes https://github.com/ember-cli/eslint-plugin-ember/issues/1268

### CODE REVIEW

**Comment by raido** [comment]
@bmish 🤞  for a review :)

File: lib/rules/no-shadow-route-definition.js [hunk 1]
```diff
@@ -71,11 +71,9 @@ module.exports = {
           return;
         }
 
-        if (
-          routeMap.has(routeInfo.route.fullPathWithGenericParams) &&
-          !isNestedRouteWithSamePath(routeInfo)
-        ) {
-          const existingRouteInfo = routeMap.get(routeInfo.route.fullPathWithGenericParams);
+        const routeLookupKey = `${routeInfo.route.blockStatementsTreePrefix}::${routeInfo.route.fullPathWithGenericParams}`;
```
Comment (raido) [inline lib/rules/no-shadow-route-definition.js:74]: For routes that are not within if else blocks, key will look like "::<path>" - we could set it "global::<path>" for easier understanding what's the contents of route map. 🤷‍♂️

**Review by bmish — APPROVED** [review]
The logic in this rule is kind of hard to follow, but this looks good based on the test case. Thanks.

**Comment by raido** [comment]
It sure has grown over time more complex than original implementation. 

Refactoring it should be done in another PR though.
